docs/topics/ingress.md

```diff
@@ -0,0 +1,226 @@
+
+This created some Kubernetes Custom Resources Definitions (CRDs), and deployed the Helm chart. Among the CRDs created is one of type `Issuer` - when an ingress resource is annotated with a specific issuer, this will request a TLS certificate from the ACME server URL from the definition below, and before expiration, an email will be sent to the address defined below - so make sure to edit the file.
+
+Let's Encrypt defines two tiers for emitting certificates - staging and production - production certificates have very strong [limit rates](https://letsencrypt.org/docs/rate-limits/), so in order to make sure your setup works, start with generating staging certificates, then once you know it works, create a production one.
+
+Below you can find the definitions for each of the tiers:
+
+```yaml
+   apiVersion: certmanager.k8s.io/v1alpha1
+   kind: ClusterIssuer
+   metadata:
+     name: letsencrypt-staging
+   spec:
+     acme:
+       # The ACME server URL
+       server: https://acme-staging-v02.api.letsencrypt.org/directory
+       # Email address used for ACME registration
+       email: user@example.com
+       # Name of a secret used to store the ACME account private key
+       privateKeySecretRef:
+         name: letsencrypt-staging
+       # Enable the HTTP-01 challenge provider
+       http01: {}
+```
+
+This is the definition for the production issuer (also make sure to change the email used):
+
+```yaml
+   apiVersion: certmanager.k8s.io/v1alpha1
+   kind: ClusterIssuer
+   metadata:
+     name: letsencrypt-prod
+   spec:
+     acme:
+       # The ACME server URL
+       server: https://acme-v02.api.letsencrypt.org/directory
+       # Email address used for ACME registration
+       email: user@example.com
+       # Name of a secret used to store the ACME account private key
+       privateKeySecretRef:
+         name: letsencrypt-prod
+       # Enable the HTTP-01 challenge provider
+http01: {}
+```
+
+> Note: `ClusterIssuer` objects work across namespaces - if you want to limit generating certificates to a single namespace, use an `Issuer` object.
+
+Now we need to create the cluster issuer objects defined above:
+
+```
+$ kubectl create -f staging-issuer.yaml 
+clusterissuer.certmanager.k8s.io/letsencrypt-staging created
+$ kubectl create -f prod-issuer.yaml 
+clusterissuer.certmanager.k8s.io/letsencrypt-prod created
+```
+
+## Putting it all together
+
+Continuing [the example from the `cert-manager` documentation](http://docs.cert-manager.io/en/latest/tutorials/acme/quick-start/index.html), let's check that everything works, and you can create an ingress with TLS - the following will create a deployment, service, and ingress using [the Kubernetes Up and Running Demo](https://github.com/kubernetes-up-and-running/kuard). Make sure to change the domain in the ingress object to the domain you own, and that you configured to use the IP of the Nginx ingress using your DNS provider (recall how for this we used `*.kube.radu.sh`):
+
+```yaml
```

Comment:
I'm wondering if we skip this kuard test/demo and move right on to the Brigade config/installation below?  I can be convinced otherwise, just wondering if trimming this would lend more focus on Brigade?

Reply:
The main reason it's there is to help people know they have everything set up correctly -- if it's your first time doing this, and you can't get a Brigade gateway with TLS, you don't really know if it's an issue with the ingress controller, `cert-manager` or Brigade -- this way, you get a clear example (not a template) of how a deployment, service, and ingress should look like.

At the same time, I agree that it's taking quite a big portion of the entire doc, so I also get your point.

Reply:
Yes, good point.  And, hey, it helps some of us (including me 😅 ) learn about kuard!